tests/external_population_test.go

```diff
@@ -0,0 +1,363 @@
+	"kubevirt.io/containerized-data-importer/tests/utils"
+)
+
+const (
+	samplePopulatorName = "sample-populator"
+	populatorGroupName  = "cdi.sample.populator"
+	populatorAPIVersion = "v1alpha1"
+	populatorKind       = "SamplePopulator"
+	populatorResource   = "samplepopulators"
+	snapshotAPIName     = "snapshot.storage.k8s.io"
+)
+
+var _ = Describe("Population tests", func() {
+	f := framework.NewFramework("population-func-test")
+
+	var (
+		fileName          string
+		expectedContent   string
+		samplePopulatorCR *unstructured.Unstructured
+	)
+
+	sampleGVR := schema.GroupVersionResource{Group: populatorGroupName, Version: populatorAPIVersion, Resource: populatorResource}
+	apiGroup := populatorGroupName
+	dataSourceRef := &corev1.TypedLocalObjectReference{
+		APIGroup: &apiGroup,
+		Kind:     populatorKind,
+		Name:     samplePopulatorName,
+	}
+
+	// If the AnyVolumeDataSource feature gate is disabled, Kubernetes drops the contents of the dataSourceRef field.
+	// We can then determine if the feature is enabled or not by checking that field after creating a PVC.
+	isAnyVolumeDataSourceEnabled := func() bool {
+		pvc := utils.NewPVCDefinition("test", "10Mi", nil, nil)
+		pvc.Spec.DataSourceRef = dataSourceRef
+		pvc, err := f.CreatePVCFromDefinition(pvc)
+		Expect(err).ToNot(HaveOccurred())
+		enabled := pvc.Spec.DataSourceRef != nil
+		err = f.DeletePVC(pvc)
+		Expect(err).ToNot(HaveOccurred())
+		return enabled
+	}
+
+	getSnapshotClassName := func() string {
+		storageclass, err := f.K8sClient.StorageV1().StorageClasses().Get(context.TODO(), f.SnapshotSCName, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		scs := &snapshotv1.VolumeSnapshotClassList{}
+		err = f.CrClient.List(context.TODO(), scs)
+		Expect(err).ToNot(HaveOccurred())
+		for _, snapshotClass := range scs.Items {
+			if snapshotClass.Driver == storageclass.Provisioner {
+				return snapshotClass.Name
+			}
+		}
+		return ""
+	}
+
+	deploySamplePopulator := func() error {
+		By("Creating Sample Populator CR")
+		fileName = fmt.Sprintf("example-%s.txt", f.Namespace.Name)
+		expectedContent = fmt.Sprintf("Hello from namespace %s", f.Namespace.Name)
+		samplePopulatorCR = &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"kind":       populatorKind,
+				"apiVersion": populatorGroupName + "/" + populatorAPIVersion,
+				"metadata": map[string]interface{}{
+					"name":      samplePopulatorName,
+					"namespace": f.Namespace.Name,
+				},
+				"spec": map[string]interface{}{
+					"fileName":     fileName,
+					"fileContents": expectedContent,
+				},
+			},
+		}
+
+		_, err := f.DynamicClient.Resource(sampleGVR).Namespace(f.Namespace.Name).Create(
+			context.TODO(), samplePopulatorCR, metav1.CreateOptions{})
+		return err
+	}
+
+	getNonCSIStorage := func() (string, bool) {
+		localSCName := "local"
+		// We first check if the default storage class lacks CSI drivers
+		if utils.DefaultStorageClassCsiDriver != nil {
```

Comment:
don't you mean == nil?
i.e. no csi driver=>return default sc

Reply:
Wow thanks for noticing :facepalm: